lib/core/macros.nim

```diff
@@ -460,25 +460,35 @@ type
     brForceOpen          ## same as brOpen, but it will always be open even
                          ## if not ambiguous (this cannot be achieved with
                          ## any other means in the language currently)
-
-proc bindSym*(ident: string | NimNode, rule: BindSymRule = brClosed): NimNode {.
-              magic: "NBindSym", noSideEffect.}
-  ## creates a node that binds `ident` to a symbol node. The bound symbol
-  ## may be an overloaded symbol.
-  ## if `ident` is a NimNode, it must have nkIdent kind.
-  ## If ``rule == brClosed`` either an ``nkClosedSymChoice`` tree is
-  ## returned or ``nkSym`` if the symbol is not ambiguous.
-  ## If ``rule == brOpen`` either an ``nkOpenSymChoice`` tree is
-  ## returned or ``nkSym`` if the symbol is not ambiguous.
-  ## If ``rule == brForceOpen`` always an ``nkOpenSymChoice`` tree is
-  ## returned even if the symbol is not ambiguous.
-  ##
-  ## experimental feature:
-  ## use {.experimental: "dynamicBindSym".} to activate it
-  ## if called from template / regular code, `ident` and `rule` must be
-  ## constant expression / literal value.
-  ## if called from macros / compile time procs / static blocks,
-  ## `ident` and `rule` can be VM computed value.
+  NimScope* = distinct ByteAddress ## scope in which to resolve `bindSym`
+
+proc bindSym*(ident: string | NimNode, rule: BindSymRule = brClosed): NimNode {.magic: "NBindSym", noSideEffect.}
+
+when defined(nimHasGetCurrentScope):
+  proc getCurrentScope*(): NimScope {. magic: "GetCurrentScope", noSideEffect.}
+    ## return opaque pointer to current scope, allowing `bindSym` to
+    ## resolve relatively to it
+  proc bindSym*(ident: string | NimNode, rule: BindSymRule = brClosed, scope: NimScope): NimNode {.magic: "NBindSym", noSideEffect.}
```

Comment:
I think scope should be first argument here. First of all, I am not a big fan of argument without default value _after_ arguments with default value. Second, I would prefer to write ``scope.bindSym("abc")`` over ``bindSym("abc", scope=scope)``.

Reply:
> First of all, I am not a big fan of argument without default value after arguments with default value

I've pushed a commit to avoid overloading `bindSym`, so that `scope` now has a default value (of 0).

Making `scope` 1st argument seems worse because it'd require to have both `bindSym(ident, rule)` (that must be kept for backward compatibility), plus `bindSym(scope, ident, rule)`.